### PR TITLE
MNTOR-2643 - detect client env consistently, and enable debug logging…

### PR DESCRIPTION
**Pull request by rhelmer**
… for Sentry

# References: 
Jira: MNTOR-2643

# Description

I noticed we're not getting performance metrics from Sentry, after consulting the docs the only thing that seems wrong potentially is the way we're checking for the environment. I'd also like to enable debug logging in Sentry so we can see if it's working correctly on dev+stage, if that's fine then we can disable that again (added FIXME comment with a link to the issue).

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by rhelmer** [comment]
I'm not sure the way we're checking the environment should really matter all that much because it just changes the sampling...

File: sentry.client.config.ts [hunk 1]
```diff
@@ -13,27 +13,26 @@ Sentry.init({
   dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,
 
   // Adjust this value in production, or use tracesSampler for greater control
-  tracesSampleRate: ["development"].includes(process.env.NODE_ENV) ? 1.0 : 0.1,
+  tracesSampleRate: ["local"].includes(getEnvironment()) ? 1.0 : 0.1,
 
   // Setting this option to true will print useful information to the console while you're setting up Sentry.
-  debug: false,
+  // FIXME set this to `false` after concluding testing, @see MNTOR-2643
+  debug: true,
```
Comment (Vinnl) [inline sentry.client.config.ts:20]: Wouldn't something like this be OK to leave in indefinitely?

```suggestion
  debug: getEnvironment() !== "production",
```

**Review by Vinnl — APPROVED** [review]
Heh, nice catch.

**Comment by rhelmer** [comment]
OK looks like we also need to enable `BrowserTracing` both in the client and on the server:
https://docs.sentry.io/platforms/javascript/configuration/integrations/browsertracing/

I don't have permission to enable it on the server side, will follow up on that bit.

**Pull request merged**